### PR TITLE
bump golang to 1.18.3

### PR DESCRIPTION
**Pull request by chrisho**
**Problem:**
Golang version upgrade

**Solution:**
Bump to golang 1.18.3

**Related Issue:**
https://github.com/harvester/harvester/issues/2315

**Test plan:**
**Validate you can develop harvester locally**
make
docker push image to your repository
The image is built successfully and can be use in a existing cluster

### CODE REVIEW

**Review by guangbochen — APPROVED** [review]
LGTM

File: go.mod [hunk 1]
```diff
@@ -1,59 +1,157 @@
...
-	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.21.0
-	k8s.io/kubectl => k8s.io/kubectl v0.21.0
-	k8s.io/kubelet => k8s.io/kubelet v0.21.0
-	k8s.io/kubernetes => k8s.io/kubernetes v1.20.2
-	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.21.0
-	k8s.io/metrics => k8s.io/metrics v0.21.0
-	k8s.io/mount-utils => k8s.io/mount-utils v0.21.0
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.21.0
+	google.golang.org/grpc => google.golang.org/grpc v1.43.0
+	k8s.io/api => k8s.io/api v0.23.0
```
Comment (futuretea) [inline go.mod:18]: seems is not the same version with harvester/harvester master
https://github.com/harvester/harvester/blob/9f6f58bc1c908d3d3c68491ebd1a56cbe954458c/go.mod#L21

File: go.mod [hunk 2]
```diff
@@ -1,59 +1,157 @@
...
 
-go 1.16
+go 1.18
 
 replace (
 	github.com/dgrijalva/jwt-go => github.com/dgrijalva/jwt-go v3.2.1-0.20200107013213-dc14462fd587+incompatible
 	github.com/docker/distribution => github.com/docker/distribution v0.0.0-20191216044856-a8371794149d
 	github.com/docker/docker => github.com/docker/docker v1.4.2-0.20200203170920-46ec8731fbce
-	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.4.1
+	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.5.5
```
Comment (futuretea) [inline go.mod:9]: seems is not the same version with harvester/harvester master

**Review by futuretea — APPROVED** [review]
LGTM

**Review by guangbochen — APPROVED** [review]

**Review by yaocw2020 — APPROVED** [review]
LGTM

**Pull request merged**